src/app.py

```diff
@@ -335,11 +335,34 @@ def query(
                 rule_author=rule.author,
                 is_global=rule.is_global,
                 is_private=rule.is_private,
+                is_degenerate=rule.parse().is_degenerate,
                 parsed=rule.parse().query,
             )
             for rule in rules
         ]
 
+    degenerate_rules = [r.name for r in rules if r.parse().is_degenerate]
+    allow_slow = db.get_mquery_config_key("query_allow_slow") == "true"
```

Comment:
Probably nitpicky, but should configuration really be in Redis? From my understanding Redis has a couple of responsibilities in this project: it's used as some sort of state management to keep the state of plugins, but also to store configuration, and there may be other uses I don't remember of the top of my head (like as a cache mechanism, possibly?)

Seems a bit cluttered. Maybe all config should be moved to a config file with a (simple) config parser implemented? Maybe something like in Karton. I quite like their [approach](https://github.com/CERT-Polska/karton/blob/master/karton/core/config.py#L152-L162) to loading booleans and such. Makes stuff like line 345 be as elegant as `config.get("query_allow_slow")` which would equate to 0 or 1

Reply:
I don't think that's nitpicky! Quite the opposite, I think that's way too large issue to fit into a pull request comment :).

> but should configuration really be in Redis

Definitely not. Nothing should be in Redis except job scheduling internals (and maybe cache) (see also #74). Sadly we have to live with Redis for now, because I don't want to introduce another database at this stage of the project.

> Maybe all config should be moved to a config file with a (simple) config parser implemented? (...) Makes stuff like line 345 be as elegant as (...)

There are actually three issues hidden in this simple question:

**1. Refactor plugin configuration handling to remove the boilerplate**

This is already an issue here: #257. This is especially pressing for mquery's config fields. 

**2. Refactor mquery's config handling using typedconfig or configparser**

You may not think of it like this, but mquery already has a config file - [`config.py`](https://github.com/CERT-Polska/mquery/blob/66fba1b51490a40ca6c2978f6eaf99f350013b9d/src/config.docker.py). The intention of this file is to be modified with the deployment-specific config (this is why it's not tracked in git, for example). Things got more complicated because of docker - to make it more idiomatic for use in containers, we have config.docker.py that just reads all config fields from the environment (this is probably how you configure mquery).

I'm not fond of this pattern, but the current config is extremely simple (just backend URL, redis URL, forgotten "job expiration minutes" field #318 and plugin configuration).
I have it in the back of my mind to migrate someday to typedconfig or configparser, but (right now) it's a low priority refactoring and I'm weary of breaking backward compatibility.

**3. Move mquery core config fields from `plugin:Mquery` in the database to the config file**

(note: You say "all config should be moved", but I assume you mean all "core" config from `plugin:Mquery`. I don't think it's feasible to move plugin config to config file.)

After reading this I was initially dismissive of the idea, thinking it's unfeasible. After thinking a bit though, looks like [all current config fields](https://github.com/CERT-Polska/mquery/blob/66fba1b51490a40ca6c2978f6eaf99f350013b9d/src/db.py#L248) are viable candidates for a config file. This may be a bit harder for config fields that will influence worker behaviour (like #297 will), but still doable.

What I like about this idea:

* Less reinventing the whell, and fewer lines of code. Config refactor (#1) solved for free.
* Better separation of concerns - redis for plugins configuration, config file for mquery's core and web config.

What I don't like about this idea:

* This will break backward compatibility and we already have a release with redis-based mquery config. I think I can stomach that since I think number of orgs that integrated mquery auth with their orgs is very limited right now :).
* Impossible to configure mquery at runtime. Not necessarily a problem, current configuration keys should not be modified lightly and probably would benefit from being less interactive.
* At this point mquery's config gets a bit complicated, so we should combined this with (#2). I definitely don't want to have three configuration sources (redis, config.py, mquery.ini), therein lies madness. This means more backward-incompatible changes.

So wrapping up, I think you're onto something, but it needs a bit more design and it's definitely out of scope for this PR 🙃. Would you mind creating a separate issue for that?